govcd/metadata.go

```diff
@@ -147,3 +148,105 @@ func addMetadata(client *Client, key string, value string, requestUri string) (T
 	return client.ExecuteTaskRequest(apiEndpoint.String(), http.MethodPut,
 		types.MimeMetaDataValue, "error adding metadata: %s", newMetadata)
 }
+
+// GetMetadata() function calls private function getMetadata() with catalogItem.client and catalogItem.CatalogItem.HREF
+// which returns a *types.Metadata struct for provided catalog item input.
+func (vAppTemplate *VAppTemplate) GetMetadata() (*types.Metadata, error) {
+	return getMetadata(vAppTemplate.client, vAppTemplate.VAppTemplate.HREF)
+}
+
+// AddMetadata() function adds metadata key, value pair provided as input and returned update VAppTemplate
+func (vAppTemplate *VAppTemplate) AddMetadata(key string, value string) (*VAppTemplate, error) {
+	task, err := vAppTemplate.AddMetadataAsync(key, value)
+	if err != nil {
+		return nil, err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("error completing add metadata for vApp template task: %#v", err)
+	}
+
+	err = vAppTemplate.Refresh()
+	if err != nil {
+		return nil, fmt.Errorf("error refreshing vApp template: %#v", err)
+	}
+
+	return vAppTemplate, nil
+}
+
+// AddMetadata() function calls private function addMetadata() with vAppTemplate.client and vAppTemplate.VAppTemplate.HREF
```

Comment:
this comment should say AddMetadataAsync

Reply:
fixed